cmd/kubectl-gadget/bcck8s/bcck8s.go

```diff
@@ -0,0 +1,235 @@
+// Copyright 2019-2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bcck8s
+
+import (
+	"context"
+	"crypto/rand"
+	"errors"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"github.com/kinvolk/inspektor-gadget/cmd/kubectl-gadget/utils"
+	"github.com/kinvolk/inspektor-gadget/pkg/k8sutil"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func BccCmd(subCommand, bccScript string, params *utils.CommonFlags, stackFlag, uniqueFlag, profileKernel, profileUser *bool) func(*cobra.Command, []string) {
```

Comment:
It's not need to have `uniqueFlag, profileKernel, profileUser` as arguments. Having a `gadgetParameters []string` argument and letting each gadget fill it according to their needs would be much more clear. 

For instance, the profile gadget will pass []string{"-f", "-d"} and also "-U" or "-K" according to profileUser / profileKernel. Other gadgets won't care about this flag as it's not important for them.

Reply:
You are right but `uniqueFlag, profileKernel, profileUser` are used in `BccCmd`, so this is why I added them.
Also, I did not want to modify a lot this function because before end of the year (I think) we will get rid of it.
So, this was more a "patch to get it work at any cost" rather than a "patch to get it work cleanly by improving things".